service/src/main/java/tds/config/services/impl/ConfigServiceImpl.java

```diff
@@ -24,16 +26,35 @@ public ConfigServiceImpl(ConfigRepository configRepository) {
 
     @Override
     public Optional<ClientTestProperty> getClientTestProperty(final String clientName, final String assessmentId) {
-        ClientTestProperty clientTestProperty = configRepository.getClientTestProperty(clientName, assessmentId);
-
-        return Optional.ofNullable(clientTestProperty);
+        return configRepository.getClientTestProperty(clientName, assessmentId);
     }
 
     @Override
     public Optional<ClientSystemFlag> getClientSystemFlag(String clientName, String auditObject) {
         List<ClientSystemFlag> clientSystemFlags = configRepository.getClientSystemFlags(clientName);
 
-        return clientSystemFlags.stream().filter(f -> f.getAuditObject().equals(auditObject))
+        return clientSystemFlags.stream()
+                .filter(f -> f.getAuditObject().equals(auditObject))
                 .findFirst();
     }
+
+    @Override
+    public Optional<Integer> getTaCheckInTimeLimit(final String clientName) {
+        Optional<TimeLimits> timeLimits =  configRepository.getTimeLimits(clientName);
+
+        if (timeLimits.isPresent()) {
+            return Optional.of(
+                    timeLimits.map(TimeLimits::getTaCheckinTimeMinutes)
+                    .orElse(0));
```

Comment:
I guess I'm struggling with some of this stuff.  If the code doesn't think it should be null then why do we allow it in the DB?  I understand legacy code stuff but if the app doesn't expect null then I don't understand why we don't enforce it so we don't need extra code.